apis/config/v1beta1/configuration_types.go

```diff
@@ -169,6 +169,12 @@ type ControllerMetrics struct {
 	// metrics will be reported.
 	// +optional
 	EnableClusterQueueResources bool `json:"enableClusterQueueResources,omitempty"`
+
+	// CertDir is the directory that contains the server key and certificate.
+	// The server key and certificate
+	// must be named tls.key and tls.crt, respectively.
+	// +optional
+	CertDir string `json:"certDir,omitempty"`
```

Comment:
Do we need this API at all? @tenzen-y was proposing we could use the default CertDir location whenever the internal cert manager is disabled. I like this approach as light-weight on the API: https://github.com/kubernetes-sigs/kueue/issues/3259#issuecomment-2578652710.

I'm ok with the CertDir as a configuration, but it seems counter-intuitive that it enables the feature.
wdyt @tenzen-y ?

Reply:
This is the recommended approach from kubebilder. We use CertDir for webhooks but I think reusing the same certificates may not be the best thing.

Reply:
https://github.com/kubernetes-sigs/kubebuilder/blob/d72fa069e3190adcf53a64a4e4daddc89cb3653e/testdata/project-v4/cmd/main.go#L87

Reply:
I see what you mean.

I could add an enum and hardcode cert path in case of cert manager. And for self serving I could use the other path.

Reply:
Resolving as our API. Hange